File: backend/product-service/src/main/java/com/points/product/entity/Product.java
package com.points.product.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * 产品实体
 * 使用UUID作为主键，提高安全性和分布式支持
 */
@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    @Id
    @Column(length = 36)
    private String id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false)
    private Integer points; // 所需积分

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Integer stock; // 库存

    @Column(length = 500)
    private String image; // 图片URL

    @Column(name = "monthly_limit", nullable = false)
    private Integer monthlyLimit; // 每月兑换限制

    @Column(nullable = false)
    private Integer status; // 0-下架 1-上架

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        if (id == null) {
            id = UUID.randomUUID().toString();
        }
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (status == null) status = 1;
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: admin-frontend/src/api/product.ts
import api from './auth'

export interface Product {
  id: string  // UUID
  name: string
  points: number
  description: string
  stock: number
  image: string
  monthlyLimit: number
  status: number
  usedThisMonth?: number
}

export const getAllProducts = async () => {
  const response = await api.get('/products/admin/all')
  return response as Product[]
}

export const updateProduct = async (id: string, data: Partial<Product>) => {
  const response = await api.put(`/products/admin/${id}`, data)
  return response as Product
}

export const updateProductStock = async (id: string, stock: number) => {
  const response = await api.put(`/products/admin/${id}/stock`, { stock })
  return response as Product
}

export const updateProductStatus = async (id: string, status: number) => {
  const response = await api.put(`/products/admin/${id}/status`, { status })
  return response as Product
}

export const createProduct = async (data: Omit<Product, 'id' | 'usedThisMonth'>) => {
  const response = await api.post('/products/admin', data)
  return response as Product
}
